pkg/controllers/vrep/replication_reconciler.go

```diff
@@ -0,0 +1,296 @@
+/*
+ (c) Copyright [2021-2024] Open Text.
+ Licensed under the Apache License, Version 2.0 (the "License");
+ You may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package vrep
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/go-logr/logr"
+	vapi "github.com/vertica/vertica-kubernetes/api/v1"
+	v1beta1 "github.com/vertica/vertica-kubernetes/api/v1beta1"
+	"github.com/vertica/vertica-kubernetes/pkg/cmds"
+	"github.com/vertica/vertica-kubernetes/pkg/controllers"
+	vdbcontroller "github.com/vertica/vertica-kubernetes/pkg/controllers/vdb"
+	verrors "github.com/vertica/vertica-kubernetes/pkg/errors"
+	"github.com/vertica/vertica-kubernetes/pkg/events"
+	vmeta "github.com/vertica/vertica-kubernetes/pkg/meta"
+	"github.com/vertica/vertica-kubernetes/pkg/names"
+	"github.com/vertica/vertica-kubernetes/pkg/vadmin"
+	"github.com/vertica/vertica-kubernetes/pkg/vadmin/opts/replicationstart"
+	"github.com/vertica/vertica-kubernetes/pkg/vk8s"
+	vrepstatus "github.com/vertica/vertica-kubernetes/pkg/vrepstatus"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	stateReplicating          = "Replicating"
+	stateSucceededReplication = "Replication successful"
+	stateFailedReplication    = "Replication failed"
+)
+
+type ReplicationReconciler struct {
+	client.Client
+	VRec           *VerticaReplicatorReconciler
+	Vrep           *v1beta1.VerticaReplicator
+	dispatcher     vadmin.Dispatcher
+	SourcePFacts   *vdbcontroller.PodFacts
+	TargetPFacts   *vdbcontroller.PodFacts
```

Comment:
How about a map[string]*Podfacts?

Reply:
I don't want to add the extra overhead of using a map when we are certain that there will be exactly 2 keys, one for source and the other for target. So I think having 2 pointers here and directly accessing pointers would be more concise.